### ISSUE TITLE
Handle missing values in hierarchy columns

### ISSUE THREAD

**Issue by caitlinch** [AngusMcLure/PoolTestR#21]
If there are missing values in the Hierarchy columns, raise an error/warning message. This issue is likely to arise when data isn't tidy i.e., when notes are placed below data in columns. 

Either:

- Raise error saying "Fix this - you have missing values"
- Raise warning saying "There were missing values in columns A B C, dropped X rows with missing values"

Potential types of missing values to consider:

- Empty strings
- NA

**Comment by AngusMcLure** [comment]
This is a good issue to fix. Might we expand this cover missing values in any of the inputs of PoolPrev and HierPoolPrev? PoolPoweR has good system for doing input checks across the various functions that might be worth checking out.

I just checked PoolPrev as an example, and it returns a somewhat cryptic error if you have any NA's in either the result or poolSize inputs. If you have an NA in one of the grouping variables, it treats NA as one of the groups.

Finally, I'm leaning towards a warning rather than error for PoolTestR.

In summary, think there should be some kind of warning/error for NA values in any of the columns of the dataset that are used by the function (result, poolSize, grouping, hierarchy). I'm happy for you to create a separate issue to address all these

**Comment by caitlinch** [comment]
See branch `test-suite`:

- Created internal functions (`CheckInputData()`, `CheckClusterVars()`, `PrepareClusterData()`) to check data. These functions are available in file `checkInputData.R`
- Implemented tests for these functions in `test-checkInputData.R`.

Features:

- Error raised when column names in the hierarchical/clustered sampling scheme are not present within the input dataframe.
- Error raised when missing values are present within the hierarchical/clustered sampling scheme columns (i.e., `""`, `NA`, `NULL`).
- Warning raised when hierarchy values are inadequately nested (e.g., when the Site column is not a unique identifier, when the Village column has the same value in multiple Regions). I don't raise an error in this case as there may be Villages with the same name in different Regions. I do output a warning message with the problem variables for each hierarchy column, so the user can check the input. 
- Outputs the input dataframe with a single added column called `PoolTestR_id`, which simply concatenates the values of the hierarchy columns i.e., for the `SimpleExampleData` this column would have the value `{Region}_{Village}_{Site}`

Hierarchical/sampling scheme columns must be input from largest to smallest - I have updated the `HierPoolPrev()` function documentation to reflect this. This information is also included in documentation for the helper function (exported to users) `PrepareClusterData()`.

**Comment by caitlinch** [comment]
TO DO:

- [x] Implement check functions in `PoolPrev()`
- [x] Document check functions in `PoolPrev()`
- [x] Implement check functions in `HierPoolPrev()`
- [x] Document check functions in `HierPoolPrev()`
